src/mimir_coordinator.py

```diff
@@ -81,49 +86,131 @@ def is_recommended(self) -> bool:
                 return False
         return True
 
-    def build_config(self, _charm_config: Dict[str, Any]) -> Dict[str, Any]:
+    def build_config(self, s3_data: _S3StorageBackend) -> Dict[str, Any]:
         """Generate shared config file for mimir.
 
         Reference: https://grafana.com/docs/mimir/latest/configure/
         """
         mimir_config: Dict[str, Any] = {
             "common": {},
-            "alertmanager": {
-                "data_dir": str(self._root_data_dir / "data-alertmanager"),
-            },
-            "compactor": {
-                "data_dir": str(self._root_data_dir / "data-compactor"),
+            "alertmanager": self._build_alertmanager_config(),
+            "alertmanager_storage": self._build_alertmanager_storage_config(),
+            "compactor": self._build_compactor_config(),
+            "ruler": self._build_ruler_config(),
+            "ruler_storage": self._build_ruler_storage_config(),
+            "blocks_storage": self._build_blocks_storage_config(),
```

Comment:
I like this refactor.
But if these methods do not receive arguments, shouldn't be properties?

Reply:
Thought of that but decided not to switch them cause I think we might be adding to the configuration of each component with future requirements, for ex [alertmanager](https://grafana.com/docs/mimir/latest/references/configuration-parameters/#alertmanager) has tons of configuration that we haven't used yet and that might require future args passed to the functions, what I'm still not sure of tho is whether the config would reach the workers with no changes required to the config fields or if there will still be some changes the worker charm will have to do like the root directory for instance in the current case